src/Microsoft.Health.Fhir.Core/Features/Operations/ConvertData/FhirServerBuilderConvertDataRegistrationExtensions.cs

```diff
@@ -29,9 +30,17 @@ private static IFhirServerBuilder AddConvertDataEngine(this IFhirServerBuilder f
             return fhirServerBuilder;
         }
 
-        private static IFhirServerBuilder AddConvertDataTemplateProvider(this IFhirServerBuilder fhirServerBuilder)
+        private static IFhirServerBuilder AddConvertDataTemplateProviders(this IFhirServerBuilder fhirServerBuilder)
         {
-            fhirServerBuilder.Services.AddSingleton<IConvertDataTemplateProvider, ContainerRegistryTemplateProvider>();
+            fhirServerBuilder.Services.AddSingleton<DefaultTemplateProvider>();
+            fhirServerBuilder.Services.AddSingleton<ContainerRegistryTemplateProvider>();
+
+            return fhirServerBuilder;
+        }
+
+        private static IFhirServerBuilder AddTemplateProviderFactory(this IFhirServerBuilder fhirServerBuilder)
+        {
+            fhirServerBuilder.Services.AddSingleton<ITemplateProviderFactory, TemplateProviderFactory>();
```

Comment:
We could move this to be after line 36. It doesn't make too much sense to me to wrap up just another method call with a new one.